src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/EphemeralProjectSnapshot.cs

```diff
@@ -16,24 +17,29 @@ internal class EphemeralProjectSnapshot : IProjectSnapshot
     private readonly HostWorkspaceServices _services;
     private readonly Lazy<RazorProjectEngine> _projectEngine;
 
-    public EphemeralProjectSnapshot(HostWorkspaceServices services, string filePath)
+    public EphemeralProjectSnapshot(HostWorkspaceServices services, ProjectKey projectKey)
     {
         if (services is null)
         {
             throw new ArgumentNullException(nameof(services));
         }
 
-        if (filePath is null)
+        if (projectKey is null)
         {
-            throw new ArgumentNullException(nameof(filePath));
+            throw new ArgumentNullException(nameof(projectKey));
         }
 
         _services = services;
-        FilePath = filePath;
+        Debug.Assert(projectKey.Id.EndsWith(".csproj", StringComparison.OrdinalIgnoreCase), "EphemeralProjectSnapshot should only be used in the legacy editor, where projects are tracked by the full path to the .csproj file");
```

Comment:
❓Is there a better way to check this? Something in HostWorkspaceServices that is unique to the legacy editor?

Reply:
This is removed in a subsequent PR, as it turns out, and its back to being a string.

Reply:
Actually, I can move this whole type to the legacy editor side. Will do that in the next PR.